modules/java_api/src/main/cpp/partial_shape.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2020-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <jni.h> // JNI header provided by JDK
+#include "openvino/openvino.hpp"
+
+#include "openvino_java.hpp"
+#include "jni_common.hpp"
+
+using namespace ov;
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_PartialShape_GetDimension(JNIEnv *env, jobject obj, jlong addr, jint index) {
+    JNI_METHOD("GetDimension",
+        PartialShape* partial_shape = (PartialShape *)addr;
+        return (jlong) &(*partial_shape)[index];
```

Comment:
```suggestion
        return (jlong) (partial_shape + index);
```

Reply:
Offsetting by `index` does not access the `index`-th Dimension in PartialShape. Reverting this and using the overloaded subscript operator for now.